DasharoPayloadPkg/Library/FmpDeviceSmmLib/FmpDeviceSmmLib.c

```diff
@@ -992,12 +1371,20 @@ FmpDeviceSetImageWithStatus (
 
   if (!AreImageBtgKeysCompatible(CurrentImage, Image, ImageSize)) {
     FreePool (CurrentImage);
-    AsciiPrint("New image is not signed with a compatible OEM Root Key, aborting update\n");
     DEBUG ((
       DEBUG_ERROR,
       "%a(): New image is not signed with a compatible OEM Root Key, aborting update\n",
       __FUNCTION__
       ));
+    ErrorString = L"Intel Boot Guard OEM Root Key mismatch";
+    if (AbortReason != NULL) {
+        *AbortReason = AllocateCopyPool(StrSize(ErrorString), ErrorString);
+    }
+
+    ShowBtgErrorPopup(CurrentImage, ImageSize, ErrorString);
+
+    FreePool(ErrorString);

```

Comment:
No need to free a string literal.

Reply:
https://github.com/Dasharo/edk2/pull/285/commits/170d6b39ccff98e61582de6a7547bc012bdce27a